src/Composer/RuntimeFile.php

```diff
@@ -0,0 +1,189 @@
+    {
+        $options = [];
+        $repo = $this->composer->getRepositoryManager()->getLocalRepository();
+
+        $packages = array_merge(
+            $repo->getPackages(),
+            [$this->composer->getPackage()]
+        );
+        foreach ($packages as $package) {
+            $packageOptions = $this->getPackageRuntimeOptions($package);
+            if (empty($packageOptions)) {
+                continue;
+            }
+            $options[$package->getName()] = $packageOptions;
+        }
+
+        return $options;
+    }
+
+    private function getRuntimeTemplateFile(): string
+    {
+        $fs = new Filesystem();
+
+        /** @var RuntimeRootPackageOptions $rootPackageOptions */
+        $rootPackageOptions = $this->getPackageRuntimeOptions(
+            $this->composer->getPackage()
+        );
+        $runtimeTemplateFile = $rootPackageOptions['template'] ?? null;
+        if ($runtimeTemplateFile === null) {
+            $runtimeTemplateFile = __DIR__ . '/atoolo_runtime.template';
+        } else {
+            if (!$fs->isAbsolutePath($runtimeTemplateFile)) {
+                $runtimeTemplateFile = $this->projectDir . '/'
+                    . $runtimeTemplateFile;
+            }
+
+            if (!is_file($runtimeTemplateFile)) {
+                throw new InvalidArgumentException(
+                    sprintf(
+                        'File "%s" defined under '
+                        . '"extra.atoolo.runtime.template"'
+                        . ' in your composer.json file not found.',
+                        $runtimeTemplateFile
+                    )
+                );
+            }
+        }
+
+        return $runtimeTemplateFile;
+    }
+
+    /**
+     * @return RuntimePackageOptions|RuntimeRootPackageOptions
+     */
+    private function getPackageRuntimeOptions(PackageInterface $package): array
+    {
+        /** @var array{
+         *     atoolo?: array{
+         *         runtime?: RuntimePackageOptions|RuntimeRootPackageOptions
+         *     }
+         * } $extra
+         */
+        $extra = $package->getExtra();
+        return $extra['atoolo']['runtime'] ?? [];
+    }
+    private function getProjectDir(): string
+    {
+
+        $vendorDir = $this->getVendorDir();
+
+        $fs = new Filesystem();
+        $projectDir = $fs->makePathRelative($this->projectDir, $vendorDir);
+        $nestingLevel = 0;
+
+        while (str_starts_with($projectDir, '../')) {
+            ++$nestingLevel;
+            $projectDir = substr($projectDir, 3);
+        }
+
+        if (!$nestingLevel) {
+            $projectDir = '__' . 'DIR__.'
+                . var_export('/' . $this->projectDir, true);
+        } else {
+            $projectDir = 'dirname(__' . "DIR__, $nestingLevel)"
+                . ('' !== $this->projectDir ? '.'
+                    . var_export('/' . $this->projectDir, true) : ''
+                );
+        }
+
+        return $projectDir;
```

Comment:
I believe we can make this a bit more concise:
```suggestion
        $dirname = $nestingLevel === 0
            ? '__DIR__.'
            : sprintf('dirname(__DIR__, %d)', $nestingLevel);
        return '' !== $this->projectDir
            ? $dirname . var_export('/' . $this->projectDir, true)
            : $dirname;
```